test/end2end_test.go

```diff
@@ -193,6 +193,16 @@ func (s *testServer) HalfDuplexCall(stream testpb.TestService_HalfDuplexCallServ
 
 const tlsDir = "testdata/"
 
+func TestDialTimeout(t *testing.T) {
+	conn, err := grpc.Dial("localhost:0", grpc.WithTimeout(5*time.Microsecond))
+	if err == nil {
+		conn.Close()
+	}
+	if err != grpc.ErrTimeout {
+		t.Fatalf("Expected %v, got %v", grpc.ErrTimeout, err)
```

Comment:
t.Fatalf("grpc.Dail(_, _) = %v, %v, want <nil>, %v", conn, err, grpc.ErrClientConnTimeout) to be consistent with other tests in this file.